app/src/main/java/org/greenstand/android/TreeTracker/managers/UserManager.kt

```diff
@@ -1,47 +1,55 @@
-        set(value) = sharedPreferences.edit().putString(FIRST_NAME_KEY, value).apply()
+        get() = preferences.getString(FIRST_NAME_KEY)
+        set(value) = preferences.edit().putString(FIRST_NAME_KEY, value).apply()
 
     var lastName: String?
-        get() = sharedPreferences.getString(LAST_NAME_KEY, null)
-        set(value) = sharedPreferences.edit().putString(LAST_NAME_KEY, value).apply()
+        get() = preferences.getString(LAST_NAME_KEY)
+        set(value) = preferences.edit().putString(LAST_NAME_KEY, value).apply()
+
+    var profilePhotoPath: String?
+        get() = preferences.getString(PROFILE_PHOTO_PATH_KEY)
+        set(value) = preferences.edit().putString(PROFILE_PHOTO_PATH_KEY, value).apply()
 
     var organization: String?
-        get() = sharedPreferences.getString(ORG_NAME_KEY, null)
-        set(value) = sharedPreferences.edit().putString(ORG_NAME_KEY, value).apply()
+        get() = preferences.getString(ORG_NAME_KEY)
+        set(value) = preferences.edit().putString(ORG_NAME_KEY, value).apply()
 
     var planterCheckinId: Long?
-        get() = sharedPreferences.getLong(ValueHelper.PLANTER_CHECK_IN_ID, -1)
-        set(value) = sharedPreferences.edit().putLong(ValueHelper.PLANTER_CHECK_IN_ID, value ?: -1).apply()
+        get() = preferences.getLong(PLANTER_CHECK_IN_ID_KEY)
+        set(value) = preferences.edit().putLong(PLANTER_CHECK_IN_ID_KEY, value ?: -1).apply()
 
     var planterInfoId: Long?
-        get() = sharedPreferences.getLong(ValueHelper.PLANTER_INFO_ID, -1)
-        set(value) = sharedPreferences.edit().putLong(ValueHelper.PLANTER_INFO_ID, value ?: -1).apply()
-
-    fun clearUser() {
-        sharedPreferences.edit().apply {
-            putLong(ValueHelper.TIME_OF_LAST_PLANTER_CHECK_IN_SECONDS, 0)
-            putString(ValueHelper.PLANTER_PHOTO, null)
-            putLong(ValueHelper.PLANTER_CHECK_IN_ID, -1)
-            putLong(ValueHelper.PLANTER_INFO_ID, -1)
-        }.apply()
-    }
+        get() = preferences.getLong(PLANTER_INFO_ID_KEY)
+        set(value) = preferences.edit().putLong(PLANTER_INFO_ID_KEY, value ?: -1).apply()
+
+    var lastCheckInTimeInSeconds: Long?
+        get() = preferences.getLong(LAST_CHECK_IN_TIME_IN_KEY)
+        set(value) = preferences.edit().putLong(LAST_CHECK_IN_TIME_IN_KEY, value ?: -1).apply()
+
+    fun clearUser() = preferences.clearPrefKeyUsage(BASE_KEY)
 
     companion object {
-        private const val FIRST_NAME_KEY = "FIRST_NAME_KEY"
-        private const val LAST_NAME_KEY = "LAST_NAME_KEY"
-        private const val ORG_NAME_KEY = "ORG_NAME_KEY"
+        private val BASE_KEY = PrefKeys.SESSION + PrefKey("info")
+        val PLANTER_CHECK_IN_ID_KEY = BASE_KEY + PrefKey("planter-check-in-id")
+        val PLANTER_INFO_ID_KEY = BASE_KEY + PrefKey("planter-info-id")
+        val ORG_NAME_KEY = BASE_KEY + PrefKey("organization")
+        val FIRST_NAME_KEY = BASE_KEY + PrefKey("first-name")
```

Comment:
UserPrefKey is for user unique values for that account. An example would be user A having language set to english and use B having it set to something else. PrefKey is for anything else not tied to a user.

A larger future use for UserPrefKeys will be the org flavor. A user will have specific screens enabled/disabled that will persist in shared prefs per each user.

In this case, these are session based values that are overridden per user on each login/logout.

Reply:
So, UserPrefKeys are persisting the preferences across sessions. Even when UserA logs out, his preferences stored with preferences will be retained but his sessions keys will be reset. Thank you. It will be nice to have some comments on UserPrefKey and PrefKey classes for onboarding developers.

Reply:
I'll add some docs to the code and wiki later.

Reply:
Sorry, I misspoke. UserPrefKeys are for sessions and anything else. They are simply tied to a specific user. In this case the values like first_name don't need to be tied to a user. They are reset on each logout/login. No more than 1 of those values exists at a time.